### ISSUE TITLE
Deep linking - nice looking urls

### ISSUE THREAD

**Issue by necrodome** [clusterinc/skit#8]
I was playing with the rottentomatoes demo. On the main page, there are links to different lists. I tried to provide links to these lists by using server.registerUrlArgument, but couldn't find a way.

So for example, I want /t/in-theaters to match to Home controller with in-theaters matched to a key.
In general, is there a way to use named regex params for arbitrary url matching? It would be nice if I could do server.registerUrlArgument('Home', '/t/:list_name) and use query('list_name') in Home controller.

Is there a current way to accomplish this?

**Comment by taylorhughes** [comment]
Yep, so the way you would do this is by adding a controller at eg. <code>public/t/__section__/SectionController.js</code>. You don't actually need to use registerUrlArgument() if you're okay with matching any path in __section__ — paths surrounded by underscores in "public" are kind of magic.

Then loading <code>/t/foo</code> will load SectionController, and you can access "foo" by looking at <code>this.params['__section__']</code>.

**Comment by taylorhughes** [comment]
(You could probably use a symlink to reference Home.js from <code>/t/__section__/</code>, but I'm not 100% certain that would work. Probably would. Or you could abstract the common bits to a base controller in <code>/library</code> and call into it from both Home and SectionController.)

**Comment by necrodome** [comment]
I see. Both solutions seem hacky =) Any reason not to give an option for explicit routing?server.registerUrlArgument seemed like  a nice way for out of convention routing.

**Comment by taylorhughes** [comment]
I'm resisting adding options like that, because then rather than saying "every route corresponds to a directory," you have to say "every route corresponds to a directory, OR a custom route over here" — and everything becomes more complex forever. :) It adds indirection.

The best way to accomplish this in skit would be to inherit from the same base controller in both places, and not define anything new in the new Home and SectionController classes. This is what Home.js and SectionController.js would look like:

```
var Controller = skit.platform.Controller;
var BaseSectionController = library.BaseSectionController;

module.exports = Controller.create(BaseSectionController, {});
```

BaseSectionController here is just the existing Home moved to root/library and renamed, then you add the <code>if (this.params['__section__']) {</code> logic in <code>__preload__</code> to load the right section.

That should work, does not require new options, and adds no indirection to the project.

**Comment by juanbrujo** [comment]
+1 to something like dynamic routes. I'l try @taylorhughes proposed solution.  

Any clue why rottentomatoes isn't working anymore? 

![screen shot 2015-04-09 at 17 55 31](https://cloud.githubusercontent.com/assets/362186/7076711/a3c50004-dee1-11e4-9c50-f710311b0924.png)

**Comment by taylorhughes** [comment]
@juanbrujo that would happen with `skit run rottentomatoes/demo` — that project sets up a proxy, so you need to run it with `node main.js` from the rottentomatoes dir.

**Comment by juanbrujo** [comment]
dumb me, didn't see the main.js file, tnx, 

but still an error: `ERROR PROCESSING REQUEST Error: Invalid dependency: "skit.platform.iter" in module: public.Home:js`

**Comment by taylorhughes** [comment]
@juanbrujo https://github.com/clusterinc/skit/issues/9

**Comment by juanbrujo** [comment]
yep, that's it, tnk a lot, this is smt I was looking for, and congratulations. looking forward for the future of Skit.
